### PR TITLE
Validation for record requests

### PR DESCRIPTION
**Pull request by zefciu**
The RecordRequests are now validated just like records. The validation
logic was moved to a common ancestor class

### CODE REVIEW

**Comment by ghost** [comment]
:ok:

File: powerdns/utils.py [hunk 1]
```diff
@@ -7,15 +7,82 @@
...
 
+DOMAIN_NAME_RECORDS = ('CNAME', 'MX', 'NAPTR', 'NS', 'PTR')
+
+
+# Validator for the domain names only in RFC-1035
+# PowerDNS considers the whole zone to be invalid if any of the records end
+# with a period so this custom validator is used to catch them
+
+validate_domain_name = RegexValidator(
+    r'^(\*\.)?([_A-Za-z0-9-]+\.)*([A-Za-z0-9])+$'
```
Comment (xor-xor) [inline powerdns/utils.py:45]: Please provide an example of a valid/invalid domain name as a comment.

File: powerdns/utils.py [hunk 2]
```diff
@@ -193,3 +260,60 @@ def format_recursive(template, arguments):
...
+    def clean_content_field(self):
+        """Perform a type-dependent validation of content field"""
+        if self.type == 'A':
+            validate_ipv4_address(self.content)
+        elif self.type == 'AAAA':
+            validate_ipv6_address(self.content)
+        elif self.type == 'SOA':
+            validate_soa(self.content)
+        elif self.type in DOMAIN_NAME_RECORDS:
+            validate_domain_name(self.content)
```
Comment (xor-xor) [inline powerdns/utils.py]: What about adding here an `else` clause as a fallback..?

File: powerdns/utils.py [hunk 3]
```diff
@@ -193,3 +260,60 @@ def format_recursive(template, arguments):
...
+
+    def validate_for_conflicts(self):
+        """Ensure this record doesn't conflict with other records."""
+        from powerdns.models.requests import Record
+        def check_unique(comment, **kwargs):
```
Comment (xor-xor) [inline powerdns/utils.py:296]: +1 for a nested function :)

**Comment by xor-xor** [comment]
:ok:

**Pull request merged**